service_storagefacility/app/repositories/storage/dao/MigrationDao.scala

```diff
@@ -0,0 +1,414 @@
+package repositories.storage.dao
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import com.google.inject.{Inject, Singleton}
+import migration.MigrationVerification
+import models.storage.event.EventTypeRegistry.ObsSubEvents.{
+  ObsHumidityType,
+  ObsHypoxicAirType,
+  ObsPestType,
+  ObsTemperatureType
+}
+import models.storage.event.EventTypeRegistry.TopLevelEvents._
+import models.storage.event.EventTypeRegistry._
+import models.storage.event.dto._
+import models.storage.event.{EventTypeId, EventTypeRegistry}
+import no.uio.musit.MusitResults.{MusitDbError, MusitResult, MusitSuccess}
+import no.uio.musit.models._
+import play.api.Logger
+import play.api.db.slick.DatabaseConfigProvider
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import repositories.storage.dao.MigrationDao.{AllEventsRow, AllTupleType}
+import repositories.storage.old_dao.{EventTables => OldEventTables}
+import slick.basic.DatabasePublisher
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+// TODO: This can be removed when Migration has been performed.
+
+object MigrationDao {
+  type AllEventsRow =
+    (BaseEventDto, Option[ObjectUUID], Option[MuseumId])
+
+  import java.sql.{Timestamp => JSqlTimestamp}
+
+  type AllTupleType = (
+      // base
+      Long,
+      Int,
+      JSqlTimestamp,
+      Option[String],
+      Option[Long],
+      Option[Long],
+      Option[String],
+      Option[Double],
+      Option[String],
+      Option[JSqlTimestamp],
+      // actors
+      Option[Int],
+      Option[String],
```

Comment:
Something I considered when I hit the limit of the tuple size. The way your grouping actor, places etc can be done by wrapping them into a tuple. Then it also will be a bit easier to use in the mappers.